releasing/plugin-commands-publishing/src/pack.ts

```diff
@@ -98,11 +99,13 @@ export async function handler (
     throw new PnpmError('PACKAGE_VERSION_NOT_FOUND', `Package version is not defined in the ${manifestFileName}.`)
   }
   const tarballName = `${manifest.name.replace('@', '').replace('/', '-')}-${manifest.version}.tgz`
+  const catalogResolver = resolveFromCatalog.bind(null, opts.catalogs ?? {})
   const publishManifest = await createPublishManifest({
     projectDir: dir,
     modulesDir: path.join(opts.dir, 'node_modules'),
     manifest,
     embedReadme: opts.embedReadme,
+    catalogResolver,
```

Comment:
I think it will be better if you will just pass catalogs to `createExportableManifest`. Unless you do some optimization for the case with recursive publish, that requires you to pass a resolver instead.

Reply:
I can do that. Do you have any preferences on whether we need to pass catalogs from the deprecated `@pnpm/make-dedicated-lockfile` file?

Passing in a `catalogResolver` allowed us to show an error message if the `catalog:` protocol was used.

- We could pass in an empty object for catalogs, but then users will see a confusing error about there not being a catalog entry for a dependency when it does exist.
- We could read `pnpm-workspace.yaml` in that package, but I'm not sure that's worthwhile of the deprecated package.

Reply:
I don't care about `make-dedicated-lockfile` right now.

Reply:
Gotcha. I'll pass in an empty object then.